packages/shared-react/src/hooks/addresses/useAddresses.ts

```diff
@@ -112,3 +113,27 @@ export const useFetchAddressesHashesSplitByUseFrequency = () => {
     isLoading
   }
 }
+
+export const useFetchGroupedAddressesWithEnoughAlphForGas = () => {
+  const { data: addressesBalances, isLoading: isLoadingAddressesBalances } = useFetchWalletBalancesByAddress()
+
+  const addressesWithEnoughAlphForGas = useMemo(
+    () =>
+      Object.keys(addressesBalances)
+        .filter((addressHash) => !isGrouplessAddress(addressHash)) // Groupless addresses cannot be used as input for chained txs
+        .reduce((addressesWithEnoughBalance, addressHash) => {
+          const alphBalance = addressesBalances[addressHash]?.find(({ id }) => id === ALPH.id)
+          if (alphBalance && BigInt(alphBalance.availableBalance) >= MAXIMAL_GAS_FEE) {
```

Comment:
Maybe add a todo to eventually implement dynamic fee amount fetching, like we discussed?

Reply:
Added in 91de30208